dask/bag/core.py

```diff
@@ -752,6 +759,103 @@ def write(data, filename):
         f.close()
 
 
+# private functions for from_s3
+def _get_s3_bucket(bucket_name, aws_access_key, aws_secret_key, connection,
+                   anon):
+    """Connect to s3 and return a bucket"""
+    import boto
+    if anon is True:
+        connection = boto.connect_s3(anon=anon)
+    elif connection is None:
+        connection = boto.connect_s3(aws_access_key, aws_secret_key)
+    return connection.get_bucket(bucket_name)
+
+
+# we need an unmemoized function to call in the main thread. And memoized
+# functions for the dask.
+_memoized_get_bucket = toolz.memoize(_get_s3_bucket)
+
+
+def _parse_s3_URI(bucket_name, paths):
+    # FIXME
+    # DIRTY HACK: so that urlparse won't think '?' in the URI path is for a
+    # query. We replace '?' with 'QUESTIONMARK' so it stays in the path
+    # then after urlpars gives us the path replace 'QUESTIONMARK' with '?'.
+    # So this can be made to fail if a path has 'QUESTIONMARK' in it.
+    if '?' in bucket_name:
+        bucket_name = bucket_name.replace('?', 'QUESTIONMARK')
+    o = urlparse(bucket_name)
+    # if path is specified
+    if (paths == '*') and (o.path != '' and o.path != '/'):
+        paths = o.path[1:].replace('QUESTIONMARK', '?')
+    bucket_name = o.hostname
+    return bucket_name, paths
+
+
+def _from_s3(bucket_name, keys, conn_args):
+    """
+    Takes an s3 bucket name, list of bucket key, and connection info.
+    Returns a Bag.
+    """
+    name = next(load_names)
+    dsk = {}
+    for i, k in enumerate(keys):
+        aak, ask, conn, anon = conn_args
```

Comment:
It would be nice to be able to make a dask like
```python
(func, *args, **kwargs)
```
but this is apparently not valid python.